kubernetes/template/DeploymentTemplates/script.sh

```diff
@@ -98,234 +137,325 @@ check_and_move_azurestack_configuration() {
+        dpkg --configure -a
+        apt-get -f -y install
+        apt-get update 2>&1 | tee $apt_update_output | grep -E "^([WE]:.*)|([eE]rr.*)$"
+        [ $? -ne 0  ] && cat $apt_update_output && break || \
+        cat $apt_update_output
+        if [ $i -eq $retries ]; then
+            return 1
+        else
+            sleep 30
+        fi
+    done
+    
+    echo "Executed apt-get update $i time/s"
+    wait_for_apt_locks
+}
+
+# Avoid transcient apt-install failures
+# Function taken from the AKSe's code based
+apt_get_install()
+{
+    retries=$1; wait_sleep=$2; timeout=$3;
+    shift && shift && shift
+    
+    for i in $(seq 1 $retries); do
+        wait_for_apt_locks
+        dpkg --configure -a
+        apt-get install --no-install-recommends -y ${@}
+        [ $? -eq 0  ] && break || \
+        if [ $i -eq $retries ]; then
+            return 1
+        else
+            sleep $wait_sleep
+            apt_get_update
+        fi
+    done
+    
+    echo "Executed apt-get install --no-install-recommends -y \"$@\" $i times";
+    wait_for_apt_locks
 }
 
-# Basic details of the system
+#####################################################################################
+# start
+
 log_level -i "Starting Kubernetes cluster deployment."
-log_level -i "Running  script as : $(whoami)"
+log_level -i "Running script as:  $(whoami)"
+log_level -i "System information: $(uname -a)"
 
-log_level -i "System information: $(sudo uname -a)"
-WAIT_TIME_SECONDS=20
-log_level -i "Waiting for $WAIT_TIME_SECONDS seconds for system to get into stable state."
-sleep $WAIT_TIME_SECONDS
+log_level -i "------------------------------------------------------------------------"
+log_level -i "ARM parameters"
+log_level -i "------------------------------------------------------------------------"
+log_level -i "ADMIN_USERNAME:                   $ADMIN_USERNAME"
+log_level -i "AGENT_COUNT:                      $AGENT_COUNT"
+log_level -i "AGENT_SIZE:                       $AGENT_SIZE"
+log_level -i "IDENTITY_SYSTEM:                  $IDENTITY_SYSTEM"
+log_level -i "K8S_AZURE_CLOUDPROVIDER_VERSION:  $K8S_AZURE_CLOUDPROVIDER_VERSION"
+log_level -i "MASTER_COUNT:                     $MASTER_COUNT"
+log_level -i "MASTER_DNS_PREFIX:                $MASTER_DNS_PREFIX"
+log_level -i "MASTER_SIZE:                      $MASTER_SIZE"
+log_level -i "PUBLICIP_DNS:                     $PUBLICIP_DNS"
+log_level -i "PUBLICIP_FQDN:                    $PUBLICIP_FQDN"
+log_level -i "REGION_NAME:                      $REGION_NAME"
+log_level -i "RESOURCE_GROUP_NAME:              $RESOURCE_GROUP_NAME"
+log_level -i "SSH_PUBLICKEY:                    ----"
+log_level -i "STORAGE_PROFILE:                  $STORAGE_PROFILE"
+log_level -i "TENANT_ID:                        $TENANT_ID"
+log_level -i "TENANT_SUBSCRIPTION_ID:           $TENANT_SUBSCRIPTION_ID"
+log_level -i "SPN_CLIENT_ID:                    ----"
+log_level -i "SPN_CLIENT_SECRET:                ----"
```

Comment:
what about ADFS case ?. As we don't pass secret.

Reply:
I do not think that's a problem. But I can make a change to make it explicit.